python/ray/util/sgd/torch/torch_trainer.py

```diff
@@ -618,6 +639,21 @@ def default_resource_request(cls, config):
                     extra_cpu=int(remote_worker_count * num_cpus_per_worker),
                     extra_gpu=int(int(use_gpu) * remote_worker_count))
 
+            def step(self):
+                if tune_func is not None:
+                    args = inspect.signature(tune_func)
+                    if not len(args.parameters) == 2:
+                        raise ValueError("tune_func must take in exactly 2 "
+                                         "arguments. The passed in function "
+                                         "currently takes in {} "
+                                         "args".format(
+                                             str(len(args.parameters))))
```

Comment:
this inspection can be done upon definition (instead of waiting for actors to start). Can you move it to 614?

Reply:
Done.